query-engine/connectors/sql-query-connector/src/sql_trace.rs

```diff
@@ -0,0 +1,68 @@
+use opentelemetry::trace::{SpanContext, TraceContextExt};
+use quaint::ast::{Delete, Insert, Select, Update};
+use tracing::Span;
+use tracing_opentelemetry::OpenTelemetrySpanExt;
+
+pub fn trace_parent_to_string(context: &SpanContext) -> String {
+    let trace_id = context.trace_id().to_hex();
+    let span_id = context.span_id().to_hex();
+
+    // see https://www.w3.org/TR/trace-context/#traceparent-header-field-values
+    format!("traceparent=00-{}-{}-01", trace_id, span_id)
+}
+
+pub trait SqlTraceComment: Sized {
+    fn append_trace(self, span: &Span) -> Self;
+}
+
+impl SqlTraceComment for Insert<'_> {
+    fn append_trace(self, span: &Span) -> Self {
+        let span_ctx = span.context();
```

Comment:
I think you didn't put it into the supertrait because you're lacking a type bound that allows to determine that `self` can `.comment` - what you can do instead is build a really small macro to just paste the code for the impl instead.